tests/test_iib_operations.py

```diff
@@ -33,6 +33,7 @@ def test_verify_target_settings_overwrite_index_mismatch(target_settings):
         iib_operations.verify_target_settings(target_settings)
 
 
+@mock.patch("pubtools._quay.iib_operations.pm")
```

Comment:
Why mock this?

It's just an object with a well-defined API whose state you can modify locally, with no complicated setup needed, no dependency on remote systems etc.  When you could easily test with the real object, I don't get why you want to replace it with a mock - by doing that, you open the possibility of configuring the mock incorrectly.  It seems that this did in fact happen (the return_value in this test is impossible, see earlier comment about it returning a list) which allowed the change overall to be broken, yet passing CI.

Possibly there could be a mix of mocking and not mocking across different tests, but I think you should have at least one test which actually uses it in a realistic manner.  This doesn't seem difficult, isn't it approximately this:

```python
# helper class somewhere
class FakePaths:
  @hookimpl
  def get_cert_key_paths_plugin(self, server_url):
    return ("/path/to/file.crt", "/path/to/file.key")

@pytest.fixture
def fake_paths():
    # Installs a hookimpl for cert/key for the duration of a test.
    fake_paths = FakePaths()
    pm.register(fake_paths)
    yield
    pm.unregister(fake_paths)

# now any test which accepts 'fake_paths' argument will have that hookimpl
# registered during the test
```

Then your tests would be exercising the real hook mechanism, giving you confidence that it works.
Rather than exercising a mock object that you assert should match the real hook mechanism, but in reality might not.

Reply:
Modified tests to use real hooks.